integrations/access/msteams/app.go

```diff
@@ -145,6 +150,24 @@ func (a *App) init(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 
+	a.accessMonitoringRules = accessmonitoring.NewRuleHandler(accessmonitoring.RuleHandlerConfig{
+		Client:     a.apiClient,
+		PluginName: pluginName,
+		// Map msteams.RecipientData onto the common recipient type used
+		// by the access monitoring rules watcher.
+		FetchRecipientCallback: func(ctx context.Context, name string) (*common.Recipient, error) {
+			msTeamsRecipient, err := a.bot.FetchRecipient(ctx, name)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			return &common.Recipient{
+				Name: name,
+				ID:   msTeamsRecipient.ID,
+				Kind: string(msTeamsRecipient.Kind),
+			}, nil
+		},
```

Comment:
This plugin might be very slow with AMRs and we might want to increase the timeout to more than 5 seconds. Fetching recipients installs the app for every recipient, which can take several seconds and can fail.

We mitigated this by fetching every recipient on startup, this allowed to avoid timeouts and fail fast. A safer way to do this would be to tentatively fetch recipients when loading the AMR. We would still miss the fail fast, but at least we'd avoid the timeouts.

We can merge this as-is (with an increase timeout, I'd bump to at least 15 sec) and open an issue to improve the AMR logic later if that's still an issue.

Reply:
Increased the timeout for now, would installing the app for all possible recipients when initializing the rule cache at the start and then again whenever a new one is found be what you mean for the potential fix?